src/extensions/objecttype/objecttypeext_hooks.cpp

```diff
@@ -224,18 +229,68 @@ void ObjectTypeClassExt::_Clear_Voxel_Indexes()
 
 
 /**
- *  Allow to skip the check for the MCV's ActLike.
+ *  Reimplementation of ObjectTypeClass::Who_Can_Build_Me.
  *
- *  Author: ZivDero
+ *  @author: ZivDero
  */
-DECLARE_PATCH(_ObjectTypeClass_Who_Can_Build_Me_Multi_MCV_Patch)
+BuildingClass* ObjectTypeClassExt::_Who_Can_Build_Me(bool intheory, bool needsnopower, bool legal, HouseClass* house) const
 {
-    if (RuleExtension->IsMultiMCV) {
-        JMP(0x00587C0B);
+    BuildingClass* freebuilding = nullptr;
+    BuildingClass* anybuilding = nullptr;
+    int ownable = Get_Ownable();
+
+    for (int index = 0; index < Buildings.Count(); index++) {
+        BuildingClass* building = Buildings[index];
+
+        if (!building->IsInLimbo &&
+            building->House == house &&
+            building->Class->ToBuild == RTTI &&
+            (!needsnopower || building->IsPowerOn) &&
+            building->Mission != MISSION_DECONSTRUCTION && building->MissionQueue != MISSION_DECONSTRUCTION &&
+            (!legal || building->House->Can_Build(this, true, true) > 0) &&
+            building->Class->Get_Ownable() & ownable &&
+
+            /*
+            **  Construction yards can only produce objects according to their ActLike, but not if MultiMCV is enabled.
+            */
+            (!Rule->BuildConst.Is_Present(building->Class) || RuleExtension->IsMultiMCV || 1L << building->ActLike & ownable)) {
+
+            if (RTTI == RTTI_UNITTYPE || RTTI == RTTI_INFANTRYTYPE || RTTI == RTTI_BUILDINGTYPE || RTTI == RTTI_AIRCRAFTTYPE) {
+                TechnoTypeClassExtension* type_ext = Extension::Fetch(reinterpret_cast<const TechnoTypeClass*>(this));
+                BuildingTypeClassExtension* btype_ext = Extension::Fetch(building->Class);
+
+                /*
+                **  This object doesn't allow this factory to produce it.
+                */
+                if (type_ext->BuiltAt.Count() != 0 && !type_ext->BuiltAt.Is_Present(building->Class)) continue;
+
+                /*
+                **  This factory doesn't produce this kind of object.
+                */
+                if (btype_ext->IsExclusiveFactory && !type_ext->BuiltAt.Is_Present(building->Class)) continue;
```

Comment:
Isn't `!type_ext->BuiltAt.Is_Present(building->Class)` already covered by the check on line 265?

In case that check is not executed, `type_ext->BuiltAt.Count()` must be `0`, in which case `BuiltAt` can't contain the building either.